src/ORM/Rule/ExistsIn.php

```diff
@@ -129,4 +146,22 @@ protected function _fieldsAreNull($entity, $source)
         }
         return $nulls === count($this->_fields);
     }
+
+    /**
+     * Check whether there are nullable nulls in at least one part of the foreign key.
+     *
+     * @param \Cake\Datasource\EntityInterface $entity The entity to check.
+     * @param \Cake\ORM\Table $source The table to use schema from.
+     * @return bool
+     */
+    protected function _checkPartialSchemaNulls($entity, $source)
+    {
+        $schema = $source->schema();
+        foreach ($this->_fields as $field) {
+            if ($schema->isNullable($field) && $entity->get($field) === null) {
+                return true;
+            }
```

Comment:
@ionas First of all, I am OK with adding this feature. But I think @markstory has a point here. Not only 3 column primary keys, but also 2 column primary keys would cause an issue, if one column is nullable and the other one is not nullable. For example:

```sql
CREATE TABLE t1 (
    x int NOT NULL,
    y int NOT NULL,
    PRIMARY KEY (x, y)
);

CREATE TABLE t2 (
    x int NULL, -- Nulltable
    y int NOT NULL, -- Not nullable
    FOREIGN KEY(x, y) REFERENCES t1(x, y)
);

INSERT INTO t1 VALUES (1, 1);

-- OK. Your rule would pass, and the insertion would succeed :)
INSERT INTO t2 VALUES (NULL, 9999);

-- NG. Your rule would pass, but the insertion would fail :(
INSERT INTO t2 VALUES (NULL, NULL);
```

How about returning `false` if passing null into a column that has been declared `NOT NULL`?